### ISSUE TITLE
Review use of v-if vs v-show in Vue components

### ISSUE THREAD

**Issue by balmas** [alpheios-project/components#366]
This issue is to request a review of use of v-if vs v-show in the Vue components to identify potential performance issues, as seen in #360

From Slack discussion with @kirlat 

>... `v-if` triggers content initialization (including `mounted` callback for UI components) every time condition changes. That will also call all functions that are used to render values and may result in recalculation of computed props. It can be expensive!
To avoid negative effect here it can be beneficial to use `v-show` instead sometimes.

**Comment by balmas** [comment]
@irina060981 to check for this in wordlist and usage examples components
@kirlat to check for this in the other components

**Comment by irina060981** [comment]
https://github.com/alpheios-project/components/pull/411
It is for word usage components

**Comment by irina060981** [comment]
I used for wordUsage and wordList this comparision of `v-if` and `v-show`
https://vuejs.org/v2/guide/conditional.html#v-if-vs-v-show

For wordList we use only one conditional rendering - when we switch between context details and wordList inside on tab. I am using `v-if` here, because 
  - we renders context for one wordItem at once
  - it could be changed on each click and passed via Vue props
  - so it will always be re-rendered anyway 

and that's why `v-show` has no sense here at all

**Comment by irina060981** [comment]
@balmas and @kirlat , are you agree with me that I have done all my part for wordUsage and wordList features and could unassign it?

**Comment by balmas** [comment]
thanks @irina060981 . Removing you from this item.

**Comment by kirlat** [comment]
Generally, it looks good, on my opinion. There are a lot of `v-if`s used in a morphology UI component. I'm not very familiar with this code, but it seems pretty complex at first look. However, I believe we do not use any large objects there that may cause excess memory consumption and performance issues. So probably we have nothing serious to worry about.

The other part that can be optimized better are inflection tables. They were created before we started to use Vue and were oriented toward direct DOM manipulation. We inserted them successfully into Vue-based layouts and they're working well, but there is still some room for streamlining there. Inflection tables will definitely benefit from even tighter integration with Vue-based layouts, on my opinion. But I'm not sure if this should be prioritized because inflection tables are working well now and there are probably more important tasks on our plate.

Please let me know what do you think.

I've also added a small update with minor optimizations of `v-if` usage in a compact panel that can be merged into master if we're not planing any other changes here.

**Comment by balmas** [comment]
ok. I am going to enter separate issues for optimizing the morphology and inflection tables components and we will tackle them when we have less to do or when they rise in priority due to observed performance issues.
